oauth2_provider/views/oidc.py

```diff
@@ -121,3 +141,166 @@ def _create_userinfo_response(self, request):
+        if scheme == "http" and application.client_type != "confidential":
+            # http is only allowed with confidential clients.
+            raise MismatchingOIDCRedirectURIError()
+        if scheme not in application.get_allowed_schemes():
+            # Redirect to scheme ??? is not permitted.
+            raise MismatchingOIDCRedirectURIError()
+        if not application.post_logout_redirect_uri_allowed(post_logout_redirect_uri):
+            # This client does not have this redirect uri registered.
+            raise MismatchingOIDCRedirectURIError()
+
+    return prompt_logout, (post_logout_redirect_uri, application)
+
+
+class RPInitiatedLogoutView(OIDCOnlyMixin, FormView):
+    template_name = "oauth2_provider/logout_confirm.html"
+    form_class = ConfirmLogoutForm
+
+    def get_initial(self):
+        return {
+            "id_token_hint": self.oidc_data.get("id_token_hint", None),
+            "logout_hint": self.oidc_data.get("logout_hint", None),
+            "client_id": self.oidc_data.get("client_id", None),
+            "post_logout_redirect_uri": self.oidc_data.get("post_logout_redirect_uri", None),
+            "state": self.oidc_data.get("state", None),
+            "ui_locales": self.oidc_data.get("ui_locales", None),
+        }
+
+    def dispatch(self, request, *args, **kwargs):
+        self.oidc_data = {}
+        return super().dispatch(request, *args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        id_token_hint = request.GET.get("id_token_hint")
+        client_id = request.GET.get("client_id")
+        post_logout_redirect_uri = request.GET.get("post_logout_redirect_uri")
+        state = request.GET.get("state")
+
+        try:
+            prompt, (redirect_uri, application) = validate_logout_request(
+                user=request.user,
+                id_token_hint=id_token_hint,
+                client_id=client_id,
+                post_logout_redirect_uri=post_logout_redirect_uri,
+            )
+        except FatalOIDCClientError as error:
+            return self.error_response(error)
+
+        if not prompt:
+            return self.do_logout(application, redirect_uri, state)
+
+        self.oidc_data = {
+            "id_token_hint": id_token_hint,
+            "client_id": client_id,
+            "post_logout_redirect_uri": post_logout_redirect_uri,
+            "state": state,
+        }
+        form = self.get_form(self.get_form_class())
+        kwargs["form"] = form
+        if application:
+            kwargs["application"] = application
+
+        return self.render_to_response(self.get_context_data(**kwargs))
+
+    def form_valid(self, form):
+        id_token_hint = form.cleaned_data.get("id_token_hint")
+        client_id = form.cleaned_data.get("client_id")
+        post_logout_redirect_uri = form.cleaned_data.get("post_logout_redirect_uri")
+        state = form.cleaned_data.get("state")
+
+        try:
+            # Validating a second time.
```

Comment:
Validating the data (for possibly a second time) is the right behaviour. But the comment is weird.